Medical record service

gmsafe opened the pull request.

gmsafe committed:
Added medical record services

gmsafe committed:
Import and Export handlers for Medical Record

Ghost-Programmer commented on opencdx-health/src/main/java/cdx/opencdx/health/repository/OpenCDXMedicalRecordRepository.java:
```diff
@@ -0,0 +1,44 @@
+@Repository
+@Observed(name = "opencdx")
+public interface OpenCDXMedicalRecordRepository extends OpenCDXRepository<OpenCDXMedicalRecordModel> {
+    //    /**
```
Remove the commented out code

gmsafe replied:
Done

Ghost-Programmer commented on opencdx-health/src/main/java/cdx/opencdx/health/service/impl/OpenCDXMedicalRecordProcessServiceImpl.java:
```diff
@@ -0,0 +1,373 @@
+
+    private final OpenCDXCurrentUser openCDXCurrentUser;
+    private final OpenCDXMedicalRecordRepository openCDXMedicalRecordRepository;
+    //private final OpenCDXMedicalRecordService openCDXMedicalRecordService;
```
Removed Commented out code

gmsafe replied:
Done

Ghost-Programmer commented on opencdx-health/src/main/java/cdx/opencdx/health/service/impl/OpenCDXMedicalRecordServiceImpl.java:
```diff
@@ -0,0 +1,252 @@
+
+        try {
+            OpenCDXIAMUserModel currentUser = this.openCDXCurrentUser.getCurrentUser();
+            this.openCDXAuditService.phiAccessed(
```
Change to PHI created

gmsafe replied:
Done

Ghost-Programmer commented on opencdx-health/src/main/java/cdx/opencdx/health/service/impl/OpenCDXVaccineServiceImpl.java:
```diff
@@ -149,6 +150,39 @@ public Vaccine getVaccineById(GetVaccineByIdRequest request) {
+        OpenCDXVaccineModel model = this.openCDXVaccineRepository.save(vaccineModel.update(request));
+        try {
+            OpenCDXIAMUserModel currentUser = this.openCDXCurrentUser.getCurrentUser();
+            this.openCDXAuditService.phiCreated(
```
PHI Updated

gmsafe replied:
Done

Ghost-Programmer reviewed (changes requested):
18 Sonarlint issues that need to be resolved as well.

gmsafe committed:
fixed sonar and review comments

Ghost-Programmer committed:
Updates

gmsafe committed:
Medical record tests

Ghost-Programmer committed:
Fix for questionnaire

Ghost-Programmer committed:
Refactor imports and improve line indentations in multiple files

The commit refactors the organization of imports in several java files to maintain the coding standards. It shifts import statements closer to their usage and moves some unnecessary import statements. Along with that, indentations and line breaks in various files have been corrected.

Ghost-Programmer committed:
Reorganize imports in health service classes

The imports in both the OpenCDXMedicalRecordProcessServiceImpl and OpenCDXMedicalRecordProcessServiceImplTest classes have been rearranged. This does not change the functionality of the program but contributes to readability and standards compliance.

Ghost-Programmer committed:
Jmeter
